js/packages/teams-ai/src/prompts/PromptSectionBase.ts

```diff
@@ -0,0 +1,101 @@
+/**
+ * @module teams-ai
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Message } from "./Message";
+import { PromptFunctions } from "./PromptFunctions";
+import { PromptSection, RenderedPromptSection } from "./PromptSection";
+import { TurnContext } from 'botbuilder';
+import { Tokenizer } from "../tokenizers";
+import { Memory } from "../MemoryFork";
+
+/**
+ * Abstract Base class for most prompt sections.
+ */
+export abstract class PromptSectionBase implements PromptSection {
+    public readonly required: boolean;
+    public readonly tokens: number;
+    public readonly separator: string;
+    public readonly textPrefix: string;
+
+    /**
+     * Creates a new 'PromptSectionBase' instance.
+     * @param tokens Optional. Sizing strategy for this section. Defaults to `auto`.
+     * @param required Optional. Indicates if this section is required. Defaults to `true`.
+     * @param separator Optional. Separator to use between sections when rendering as text. Defaults to `\n`.
+     * @param textPrefix Optional. Prefix to use for text output. Defaults to `undefined`.
```

Comment:
comment says it defaults to `undefined` but typescript defaults it to empty string